orc8r/cloud/go/services/bootstrapper/servicers/registration/registration.go

```diff
@@ -0,0 +1,102 @@
+package registration
+
+import (
+	"context"
+	"fmt"
+
+	"magma/orc8r/cloud/go/orc8r"
+	"magma/orc8r/cloud/go/serdes"
+	"magma/orc8r/cloud/go/services/bootstrapper"
+	"magma/orc8r/cloud/go/services/device"
+	"magma/orc8r/cloud/go/services/orchestrator/obsidian/models"
+	"magma/orc8r/cloud/go/services/tenants"
+	"magma/orc8r/lib/go/protos"
+
+	"github.com/go-openapi/strfmt"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+type registrationServicer struct {}
+
+func NewRegistrationServicer() (protos.RegistrationServer, error) {
+	return &registrationServicer{}, nil
+}
+
+func (rs *registrationServicer) Register(c context.Context, request *protos.RegisterRequest) (*protos.RegisterResponse, error) {
+	nonce := nonceFromToken(request.Token)
+
+	deviceInfo, err := bootstrapper.GetGatewayDeviceInfo(context.Background(), nonce)
+	if err != nil {
+		clientErr := makeErr(fmt.Sprintf("could not get device info from nonce %v: %v", nonce, err))
+		return clientErr, nil
+	}
+
+	err = registerDevice(*deviceInfo, request.Hwid, request.ChallengeKey)
+	if err != nil {
+		clientErr := makeErr(fmt.Sprintf("error registering device: %v", err))
+		return clientErr, nil
+	}
+
+	controlProxy, err := getControlProxy(deviceInfo.NetworkId)
+	if err != nil {
+		clientErr := makeErr(fmt.Sprintf("error getting control proxy: %v", err))
+		return clientErr, nil
+	}
+
+	res := &protos.RegisterResponse{
+		Response: &protos.RegisterResponse_ControlProxy{
+			ControlProxy: controlProxy,
+		},
+	}
+	return res, nil
+}
+
+func registerDevice(ti protos.GatewayDeviceInfo, hwid *protos.AccessGatewayID, challengeKey *protos.ChallengeKey) error {
+	challengeKeyBase64 := strfmt.Base64(challengeKey.Key)
+	gatewayRecord := &models.GatewayDevice{
+		HardwareID: hwid.Id,
+		Key: &models.ChallengeKey{KeyType: challengeKey.KeyType.String(), Key: &challengeKeyBase64},
+	}
+	err := device.RegisterDevice(context.Background(), ti.NetworkId, orc8r.AccessGatewayRecordType, hwid.Id, gatewayRecord, serdes.Device)
+	return err
+}
+
+func getControlProxy(networkID string) (string, error) {
```

Comment:
I'm fine with making this a bootcamp task, but before we merge this let's ensure the ticket is created and linked here with a comment 🙂